Add "Google Search" Extension

mblode opened the pull request:
## Description

This is a new extensions that adds "Google search" extensions

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/7183998/148449524-2558f2dc-9ae9-4f90-a381-ffebc175a1e8.mp4

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

peteschaffner commented:
@mblode Thank you so much for this extension!

I just wanted to mention that it will fail, however, if you have the "Limit IP Address Tracking" setting enabled in System Preferences > Network. It might be worth mentioning in a README or something.

mblode commented:
@peteschaffner thanks for that. I've updated the Extension and added that as a note in the readme.

I've also add search history as a listing when the user hasn't typed anything which is handled via local storage. Shout out to Ayush for the suggestion. The history can be cleared and individual results can be removed

pernielsentikaer commented on extensions/google-search/package.json:
```diff
@@ -0,0 +1,44 @@
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search Google or type a URL",
+      "subtitle": "Google",
```
```suggestion
      "title": "Google Search",
```
Normally we use the subtitle, but in this case it doesn't make sense

mblode replied:
Fixed 👍

pernielsentikaer commented on extensions/google-search/src/utils/handleResults.ts:
```diff
@@ -0,0 +1,63 @@
+
+export async function getSearchResults(searchText: string, signal: AbortSignal): Promise<SearchResult[]> {
+  const response = await fetch(
+    `http://suggestqueries.google.com/complete/search?hl=en-us&output=chrome&q=${encodeURIComponent(searchText)}`,
```
```suggestion
    `https://suggestqueries.google.com/complete/search?hl=en-us&output=chrome&q=${encodeURIComponent(searchText)}`,
```

mblode replied:
Fixed 👍

pernielsentikaer reviewed (changes requested):
Hi @mblode 👋 

Thanks for your contribution, exciting extension! 🔥 

I left a few suggestions and have a single thing I noticed, It seems like there is something with the encoding in some occasions

![xvO6BV9lS0BfsCOL](https://user-images.githubusercontent.com/38033107/148701871-942e9e28-799d-4d21-8d42-78e734369b2e.png)

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

peteschaffner commented:
> @peteschaffner thanks for that. I've updated the Extension and added that as a note in the readme. Also I'm a massive fan of Tower! Best Git GUI and I've been using it for years ❤️

Great to hear Tower has been serving you well for so many years!!

BTW your extension has inspired me to [eventually] create a similar one for DuckDuckGo :)

mblode commented:
Note: I'll split this out into 2 pull requests tomorrow morning and respond to the feedback

mblode commented:
@pernielsentikaer I've resolved the issues you've raised:

- Fixed the text encoding issues by using Latin-1 (`iso-8859-1`) instead of UTF-8 for the results. E.g. "æ" will encode correctly 
- Set it to fetch from `https` instead of `http`.
- Updated the command title to "Google Search" and removed the subtitle

pernielsentikaer reviewed (approved):
Hi @mblode 👋 

MUCH better, I can't find anything else that I would like to have changed before accepting it. 

I'm going to accept it and then will my colleague do a final review before it's finally merged.

lukerooneyio commented:
@mblode Hi Matt, would it be possible to have the extension sort links at the top? I'd find it more useful such as the cases I've shown below. Besides that, thank you for this extension. This is what I wanted from Alfred years ago!!


<img width="862" alt="Screen Shot 2022-01-13 at 10 47 55 PM" src="https://user-images.githubusercontent.com/53066184/149463803-9fcd75dd-c59e-41c8-8a1a-295cb11a6946.png">
<img width="862" alt="Screen Shot 2022-01-13 at 10 48 04 PM" src="https://user-images.githubusercontent.com/53066184/149463807-d14269d5-9052-455a-88b2-8670d9cad20b.png">
e

KnifeFed commented:
Would it be possible to implement something akin to tab completion? I.e. the ability to choose "raycast google search" here and then continue typing to add e.g. "extension":

![CleanShot 2022-01-30 at 01 26 40@2x](https://user-images.githubusercontent.com/85502276/151682153-6f594ef3-d0ba-469c-9024-7e332b94717a.png)
